frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-root-page/feature-flag-root-page-content/feature-flag-root-section-card/feature-flag-root-section-card-table/feature-flag-root-section-card-table.component.ts

```diff
@@ -29,15 +31,37 @@ import { CommonStatusIndicatorChipComponent } from '../../../../../../../../shar
     RouterModule,
     DatePipe,
     CommonStatusIndicatorChipComponent,
+    SharedModule,
   ],
   templateUrl: './feature-flag-root-section-card-table.component.html',
   styleUrl: './feature-flag-root-section-card-table.component.scss',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class FeatureFlagRootSectionCardTableComponent {
+export class FeatureFlagRootSectionCardTableComponent implements OnInit, OnDestroy {
   @Input() dataSource$: MatTableDataSource<FeatureFlag[]>;
   @Input() isLoading$: Observable<boolean>;
 
+  isAllFeaturFlagFetched = false;
+  isAllFeatureFlagsFetchedSub = new Subscription();
+
+  constructor(private featureFlagsService: FeatureFlagsService) {}
+
+  ngOnInit(): void {
+    this.isAllFeatureFlagsFetchedSub = this.featureFlagsService.isAllFlagsFetched$.subscribe(
+      (value) => (this.isAllFeaturFlagFetched = value)
```

Comment:
I know this is how we have done it elsewhere, but this is a pattern we should make us pause. Any time we're subscribing to something from the store$ directly in the component just to "pull out" the latest value and save local copies of it, we are creating places for state-management bugs to creep in. Usually there's a good way to make these subscriptions go away.

Here it should be easy to refactor I think: the component doesn't need to know anything about what happens after it dispatches the fetchFlagsOnScroll action. It can always just dispatch `fetchFlagsOnScroll` and let the pre-existing filters in the `fetchFeatureFlags$` effect be where we check `isAllFlagsFetched$`. That effect itself is looking a little crazy, but that's a different story, can you try checking if allFlagsFetched in the effect instead of the component?